Projects/UOContent/Mobiles/AI/BaseAI.cs

```diff
@@ -328,596 +365,450 @@ public virtual bool HandlesOnSpeech(Mobile from)
+            }
+            else if (from.HasTrade)
+            {
+                from.SendLocalizedMessage(1070947);
+                // 1070947: You cannot friend a pet with a trade pending
+            }
+            else
+            {
+                BeginPickTarget(from, OrderType.Friend);
+            }
+        }
+    }
+    
+    private void HandleReleaseCommand(Mobile from, bool isOwner, string speech)
+    {
+        if (isOwner && WasNamed(speech) && m_Mobile.CheckControlChance(from))
+        {
+            if (!m_Mobile.Summoned)
+            {
+                from.SendGump(new ConfirmReleaseGump(from, m_Mobile));
+            }
+            else
+            {
+                m_Mobile.ControlTarget = null;
+                m_Mobile.ControlOrder = OrderType.Release;
+            }
+        }
+    }
+    
+    private void HandleTransferCommand(Mobile from, bool isOwner, string speech)
+    {
+        if (isOwner && !m_Mobile.IsDeadPet && WasNamed(speech) && m_Mobile.CheckControlChance(from))
+        {
+            if (m_Mobile.Summoned || m_Mobile is GrizzledMare)
+            {
+                from.SendLocalizedMessage(1005487);
+                // 1005487: You cannot transfer ownership of a summoned creature.
+            }
+            else if (from.HasTrade)
+            {
+                from.SendLocalizedMessage(1010507);
+                // 1010507: You cannot transfer a pet with a trade pending
+            }
+            else
+            {
+                BeginPickTarget(from, OrderType.Transfer);
+            }
+        }
+    }
+    
+    private void HandleGMCommands(SpeechEventArgs e)
+    {
+        if (m_Mobile.Debug)
+        {
+            m_Mobile.DebugSay("It is from a GM");
+        }
+    
+        if (m_Mobile.FindMyName(e.Speech, true))
+        {
+            foreach (var word in e.Speech.Split(' '))
+            {
+                if (word.InsensitiveEquals("obey"))
                 {
-                    m_Mobile.OnActionInteract();
-                    return DoActionInteract();
+                    m_Mobile.SetControlMaster(e.Mobile);
+    
+                    if (m_Mobile.Summoned)
+                    {
+                        m_Mobile.SummonMaster = e.Mobile;
+                    }
+    
+                    return;
                 }
+            }
+        }
+    }
 
-            case ActionType.Backoff:
-                {
-                    m_Mobile.OnActionBackoff();
-                    return DoActionBackoff();
-                }
+    public virtual bool Think()
+    {
+        if (m_Mobile?.Deleted != false || m_Mobile.Map == null)
+        {
+            return false;
+        }
 
-            default:
-                {
-                    return false;
-                }
+        if (CheckFlee())
+        {
+            return true;
         }
+    
+        var actionHandlers = new Dictionary<ActionType, Func<bool>>
```

Comment:
Let's not make a dictionary hot. I don't think we gain anything over a switch statement.